### PR TITLE
[5.2] Removing deprecated access to User object attributes

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
This PR removes almost all cases where we are accessing attributes of the user object through the deprecated CMSObject::get() method. All these can be replaced by simply accessing the attribute directly. This change is based on a check with phpstan.

### Testing Instructions
Codereview

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by sandewt** [comment]
I have tested this item :white_check_mark: successfully on eefbe1abf2ee77c961f769c412186d41569a86ec<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43268">issues.joomla.org/tracker/joomla-cms/43268</a>.</sub>

**Review by rdeutz — APPROVED** [review]

File: libraries/src/Application/CMSApplication.php [hunk 1]
```diff
@@ -368,7 +368,7 @@ public function execute()
      */
     protected function checkUserRequireReset($option, $view, $layout, $tasks)
     {
-        if (Factory::getUser()->get('requireReset', 0)) {
+        if (Factory::getUser()->requireReset) {
```
Comment (pe7er) [inline libraries/src/Application/CMSApplication.php]: @Hackwar please change to

if (Factory::getApplication()->getIdentity()->requireReset) {

File: libraries/src/Application/AdministratorApplication.php [hunk 2]
```diff
@@ -410,7 +410,7 @@ protected function render()
         $rootUser = $this->get('root_user');
 
         if (property_exists('\JConfig', 'root_user')) {
-            if (Factory::getUser()->get('username') === $rootUser || Factory::getUser()->id === (string) $rootUser) {
+            if (Factory::getUser()->username === $rootUser || Factory::getUser()->id === (string) $rootUser) {
```
Comment (pe7er) [inline libraries/src/Application/AdministratorApplication.php]: @Hackwar please change to

if (Factory::getApplication()->getIdentity()->username === $rootUser || Factory::getApplication()->getIdentity()->id === (string) $rootUser) {

File: libraries/src/Application/CMSApplication.php [hunk 3]
```diff
@@ -1065,7 +1065,7 @@ protected function render()
 
         $caching = false;
 
-        if ($this->isClient('site') && $this->get('caching') && $this->get('caching', 2) == 2 && !Factory::getUser()->get('id')) {
+        if ($this->isClient('site') && $this->get('caching') && $this->get('caching', 2) == 2 && !Factory::getUser()->id) {
```
Comment (pe7er) [inline libraries/src/Application/CMSApplication.php]: @Hackwar please change to

if ($this->isClient('site') && $this->get('caching') && $this->get('caching', 2) == 2 && !Factory::getApplication()->getIdentity()->id) {

**Comment by pe7er** [comment]
Thanks @Hackwar !

**Pull request merged**